src/ucp/proto/proto_am.inl

```diff
@@ -25,16 +25,16 @@ typedef void (*ucp_req_complete_func_t)(ucp_request_t *req, ucs_status_t status)
 
 
 static UCS_F_ALWAYS_INLINE ucs_status_t
-ucp_am_copy_header_to_internal_buf(ucp_request_t *req)
+ucp_am_copy_header_to_internal_buf(ucp_request_t *req, ucs_status_t status)
 {
+    unsigned mask = UCP_AM_SEND_FLAG_COPY_HEADER | 
+                          UCP_AM_SEND_PRIV_FLAG_HEADER_COPIED;
+    unsigned flags      = req->send.msg_proto.am.flags & mask;
```

Comment:
wrong alignment, or even can be simplified similar to https://github.com/openucx/ucx/pull/8173#discussion_r888706357

Reply:
Hi. I think I fix the alignment. check last commit.
Regarding simplified will do.